src/util/http_utils.js

```diff
@@ -201,12 +201,16 @@ function read_request_body(req, options) {
     });
 }
 
+function parse_xml_to_js(body, xml_options) {
+    return P.fromCallback(callback => xml2js.parseString(body, xml_options, callback));
+}
```

Comment:
If we are already changing this code then lets change from Blubird to native promises with util.promisify:
```suggestion
const parse_xml_to_js = util.promisify(xml2js.parseString);
```